bokehjs/src/lib/models/widgets/date_picker.ts

```diff
@@ -1,96 +1,75 @@
     super.connect_signals()
-    this.connect(this.model.change, () => this.render())
+
+    const {value, min_date, max_date, disabled_dates, enabled_dates, position, inline} = this.model.properties
+    this.connect(value.change, () => this._set("defaultDate", value.value()))
+    this.connect(min_date.change, () => this._set("minDate", min_date.value()))
+    this.connect(max_date.change, () => this._set("maxDate", max_date.value()))
+    this.connect(disabled_dates.change, () => this._set("disable", disabled_dates.value()))
+    this.connect(enabled_dates.change, () => this._set("enable", enabled_dates.value()))
+    this.connect(position.change, () => this._set("position", position.value()))
+    this.connect(inline.change, () => this._set("inline", inline.value()))
+
   }
 
   render(): void {
     if (this._picker != null)
-      this._picker.destroy()
+      return
 
     super.render()
 
     this.input_el = input({type: "text", class: bk_input, disabled: this.model.disabled})
     this.group_el.appendChild(this.input_el)
-
-    this._picker = new Pikaday({
-      field: this.input_el,
-      defaultDate: this._unlocal_date(new Date(this.model.value)),
-      setDefaultDate: true,
-      minDate: this.model.min_date != null ? this._unlocal_date(new Date(this.model.min_date)) : undefined,
-      maxDate: this.model.max_date != null ? this._unlocal_date(new Date(this.model.max_date)) : undefined,
-      onSelect: (date) => this._on_select(date),
+    this._picker = flatpickr(this.input_el, {
+      defaultDate: this.model.value,
+      minDate: this.model.min_date,
+      maxDate: this.model.max_date,
+      inline: this.model.inline,
+      position: this.model.position,
+      disable: _convert_date_list(this.model.disabled_dates),
+      enable: _convert_date_list(this.model.enabled_dates),
+      onChange: (selected_dates, date_string, instance) => this._on_change(selected_dates, date_string, instance),
     })
-
-    this._root_element.appendChild(this._picker.el)
-  }
-
-  _unlocal_date(date: Date): Date {
-    //Get the UTC offset (in minutes) of date (will be based on the timezone of the user's system).
-    //Then multiply to get the offset in ms.
-    //This way it can be used to recreate the user specified date, agnostic to their local systems's timezone.
-    const timeOffsetInMS = date.getTimezoneOffset() * 60000
-    date.setTime(date.getTime() - timeOffsetInMS)
-
-    const datestr = date.toISOString().substr(0, 10)
-    const tup = datestr.split('-')
-    return new Date(Number(tup[0]), Number(tup[1])-1, Number(tup[2]))
   }
 
-  _on_select(date: Date): void {
-    // Always use toDateString()!
-    // toString() breaks the websocket #4965.
-    // toISOString() returns the wrong day (IE on day earlier) #7048
-    // XXX: this should be handled by the serializer
-    this.model.value = date.toDateString()
+  _on_change(_selected_dates: any, date_string: string, _instance: flatpickr.Instance): void {
```

Comment:
https://github.com/flatpickr/flatpickr/blob/master/src/types/options.ts#L11 to get rid of the `any` type.

Reply:
@mattpap the only way I can get those types to import is to do
```
import {Options as FlatpickrOptions, DateLimit} from "flatpickr/dist/types/options"
``` 
which does seem safe/good (importing from dist). Tried installing `@types/flatpickr` but trying to import that also yields an error. The only thing exported from standard `flatpickr` seems to be the `flatpickr` instance.

Reply:
This is a common mistake that library author make, i.e. not exporting publicly enough type information. Let's leave it for now.